cmake/release_dependencies.cmake

```diff
@@ -22,12 +22,12 @@ FetchContent_Declare( hana-adapter
 
 FetchContent_Declare( header-utilities
     GIT_REPOSITORY  https://github.com/njoy/header-utilities
-    GIT_TAG         04f988c2b576630ef10fc0c3838c915f4a6e15d6
+    GIT_TAG         cc2610fee15e255c151e8e22aca1e8b3d1a96b39
```

Comment:
At some point we will want to run @nathangibson14 's scripts to generate these hashs, but it's not important right now.